packages/effect/src/internal/stream.ts

```diff
@@ -7565,6 +7566,34 @@ export const zipLatest = dual<
   ): Stream.Stream<[A, A2], E2 | E, R2 | R> => pipe(self, zipLatestWith(that, (a, a2) => [a, a2]))
 )
 
+/** @internal */
+export type ZipLatestAllValues<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = {
+  [K in keyof T]: T[K] extends Stream.Stream<infer U> ? U : never
+}
+/** @internal */
+export type ZipLatestAllErrors<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = T extends
+  Array<Stream.Stream<unknown, infer E, unknown>> ? E : never
+/** @internal */
+export type ZipLatestAllContext<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = T extends
+  Array<Stream.Stream<unknown, unknown, infer R>> ? R : never
+
+export const zipLatestAll = <T extends Array<Stream.Stream<unknown>>>(
+  ...streams: T
+): Stream.Stream<ZipLatestAllValues<T>, ZipLatestAllErrors<T>, ZipLatestAllContext<T>> => {
+  if (Arr.isTupleOf(streams, 1)) {
+    return pipe(streams[0], map((x) => [x] as ZipLatestAllValues<T>))
+  }
+  if (Arr.isTupleOfAtLeast(streams, 2)) {
+    const [head, ...tail] = streams
+    return zipLatestWith(
+      head,
+      zipLatestAll(...tail),
+      (first, second) => [first, ...second] as ZipLatestAllValues<T>
+    )
+  }
+  return empty
+}
```

Comment:
I think just using .length is easier to read and maintain.

```suggestion
): Stream.Stream<ZipLatestAllValues<T>, ZipLatestAllErrors<T>, ZipLatestAllContext<T>> => {
  if (streams.length === 0) {
    return empty
  } else if (streams.length === 1) {
    return map(streams[0], (x) => [x] as ZipLatestAllValues<T>))
  }
  const [head, ...tail] = streams
  return zipLatestWith(
    head,
    zipLatestAll(...tail),
    (first, second) => [first, ...second] as ZipLatestAllValues<T>
  )
}
```

Reply:
If you are checking only length of the array you will not get type narrowing, so 

```ts
if(streams.length === 1) {
  streams[0]
  // ^? number | undefined
}
```

My suggestion is 

```ts
if(Preficate.isTupleOf(streams, 1)) {
  streams[0]
  // ^? number
}
```

My PR with `isTupleOf` has already been merged into `next-minor` (https://github.com/Effect-TS/effect/commit/3182c89e57c87c18dbe66271149ca6a23a5b3b1b), so i think it's a perfect solution here:)